### PR TITLE
Solver

### PR DESCRIPTION
**Pull request by barakadanny**
In this PR 

- I use TDD to develop the platform

### CODE REVIEW

**Review by m05e5 — CHANGES_REQUESTED** [review]
.

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+describe Solver do
+  context 'Testing Solver Class' do
+    it 'Should return the factorial of number passed to factorial method' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eq(120)
+      expect(solver.factorial(6)).to eq(720)
+      expect(solver.factorial(7)).to eq(5040)
+    end
+  end
```
Comment (m05e5) [inline spec/solver_spec.rb]: - [ ] Hi you really did a great job. As a requirement, please kindly test your factorial with a negative number and it should return an exception. If possible also test it with `0` and it should return `1` as a result

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @barakadanny  and @kidde60 , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (barakadanny) [reply spec/solver_spec.rb]: @m05e5 Thank you for the feedback,
the change has been made.

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions from the previous CR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**